core/templates/dev/head/i18n.js

```diff
@@ -90,6 +90,11 @@ oppia.controller('I18nFooter', [
         $http.put(siteLanguageUrl, {
           site_language_code: $scope.currentLanguageCode
         });
+      } else {
+        // This is a workaround for the library page as the base url
```

Comment:
Ah, sorry, I meant something a bit different. How about:

```
// $translate.use() sets a cookie for the translation language, but does so using the
// page's base URL as the cookie path. However, the base URL is modified in pages
// like /library, thus causing the cookie path to change; in such cases, the user's
// preferences are not picked up by other pages. To avoid this, we manually set
// the cookie using the '/' path each time a non-logged-in user changes their
// site language. 
```

I think this provides the full context (i.e. a bit more info on the cookie path and what's really going on here) that would be useful to devs.